### PR TITLE
bpo-33187: Fix document ElementInclude (XInclude) support in ElementTree issue

### PR DESCRIPTION
**Pull request by bansalanjali2512**
https://bugs.python.org/issue33187

### CODE REVIEW

**Review by eamanu — APPROVED** [review]

File: Doc/library/xml.etree.elementtree.rst [hunk 1]
```diff
@@ -638,6 +638,83 @@ Functions
...
+XInclude support
+----------------
+
+This module provides limited support for
+`XInclude directives <https://www.w3.org/TR/xinclude/>`_ , via the **ElementInclude** helper module.  This module can be used to insert subtrees and text strings into element trees, based on information in the tree.
+
+Example
+^^^^^^^
+
+Here's an example that demonstrates use of the XInclude module. To include a XML document in the current document, use the ``{http://www.w3.org/2001/XInclude}include`` element and set the **parse** attribute to ``“xml”``, and use the **href** attribute to specify the document to include.::
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: "an XML document". The pronunciation of XML ("ex-em-ell") starts with a vowel.

File: Doc/library/xml.etree.elementtree.rst [hunk 2]
```diff
@@ -638,6 +638,83 @@ Functions
...
+The result might look something like: ::
+
+   <document xmlns:xi="http://www.w3.org/2001/XInclude">
+   Copyright (c) 2003.
+   </document>
+
+Reference
+---------
+
+.. _elementtree-functions:
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: Rather `elementinclude-functions`, I'd say.

File: Doc/library/xml.etree.elementtree.rst [hunk 3]
```diff
@@ -638,6 +638,83 @@ Functions
...
+
+Functions
+^^^^^^^^^
+
+.. function:: xml.etree.ElementInclude.default_loader( href, parse, encoding = None)
+
+   This loader reads an included resource from disk.  *href* is a resource
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: It isn't really defined anywhere what a `loader` is or does exactly.

File: Doc/library/xml.etree.elementtree.rst [hunk 4]
```diff
@@ -638,6 +638,83 @@ Functions
...
+   reference.  *parse* is for parse mode either "xml" or "text".  *encoding*
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: "resource reference" is not a common term. Why not simply write "URL"?

File: Doc/library/xml.etree.elementtree.rst [hunk 5]
```diff
@@ -638,6 +638,83 @@ Functions
...
+.. _elementtree-functions:
+
+Functions
+^^^^^^^^^
+
+.. function:: xml.etree.ElementInclude.default_loader( href, parse, encoding = None)
+
+   This loader reads an included resource from disk.  *href* is a resource
+   reference.  *parse* is for parse mode either "xml" or "text".  *encoding* 
+   is an optional text encoding.  If not given, none is used.  Returns the
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: "If not given, none is used." – then, what happens with the data?

File: Doc/library/xml.etree.elementtree.rst [hunk 6]
```diff
@@ -638,6 +638,83 @@ Functions
...
+Functions
+^^^^^^^^^
+
+.. function:: xml.etree.ElementInclude.default_loader( href, parse, encoding = None)
+
+   This loader reads an included resource from disk.  *href* is a resource
+   reference.  *parse* is for parse mode either "xml" or "text".  *encoding* 
+   is an optional text encoding.  If not given, none is used.  Returns the 
+   expanded resource.  If the parse mode is ``"xml"``, this is an ElementTree 
+   instance.  If the parse mode is "text", this is an Unicode string.  If the
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: "a Unicode string". "unicode" is pronounced with the consonant "y" at the beginning, like you-nee-code.

File: Doc/library/xml.etree.elementtree.rst [hunk 7]
```diff
@@ -638,6 +638,83 @@ Functions
...
+   reference.  *parse* is for parse mode either "xml" or "text".  *encoding* 
+   is an optional text encoding.  If not given, none is used.  Returns the 
+   expanded resource.  If the parse mode is ``"xml"``, this is an ElementTree 
+   instance.  If the parse mode is "text", this is an Unicode string.  If the 
+   loader fails, it can return None or raise an **IOError** exception.
+
+
+.. function:: xml.etree.ElementInclude.include( elem, loader = None)
+
+   This expand XInclude directives.  *elem* is the root element.  *loader* is
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: "This function expands XInclude directives."

File: Doc/library/xml.etree.elementtree.rst [hunk 8]
```diff
@@ -638,6 +638,83 @@ Functions
...
+   is an optional text encoding.  If not given, none is used.  Returns the 
+   expanded resource.  If the parse mode is ``"xml"``, this is an ElementTree 
+   instance.  If the parse mode is "text", this is an Unicode string.  If the 
+   loader fails, it can return None or raise an **IOError** exception.
+
+
+.. function:: xml.etree.ElementInclude.include( elem, loader = None)
+
+   This expand XInclude directives.  *elem* is the root element.  *loader* is 
+   an optional resource loader.  If omitted, it defaults to default_loader.
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: A link reference to `default_loader` would be nice here.

File: Doc/library/xml.etree.elementtree.rst [hunk 9]
```diff
@@ -638,6 +638,83 @@ Functions
...
+
+
+.. function:: xml.etree.ElementInclude.include( elem, loader = None)
+
+   This expand XInclude directives.  *elem* is the root element.  *loader* is 
+   an optional resource loader.  If omitted, it defaults to default_loader.  
+   If given, it should be a callable that implements the same interface as 
+   default_loader.  Returns the expanded resource.  If the parse mode is 
+   ``"xml"``, this is an ElementTree instance.  If the parse mode is "text", 
+   this is an Unicode string.  If the loader fails, it can return None or
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: "a Unicode string"

File: Doc/library/xml.etree.elementtree.rst [hunk 10]
```diff
@@ -638,6 +638,83 @@ Functions
...
+
+.. function:: xml.etree.ElementInclude.include( elem, loader = None)
+
+   This expand XInclude directives.  *elem* is the root element.  *loader* is 
+   an optional resource loader.  If omitted, it defaults to default_loader.  
+   If given, it should be a callable that implements the same interface as 
+   default_loader.  Returns the expanded resource.  If the parse mode is 
+   ``"xml"``, this is an ElementTree instance.  If the parse mode is "text", 
+   this is an Unicode string.  If the loader fails, it can return None or 
+   raise an **IOError** exception.
```
Comment (scoder) [inline Doc/library/xml.etree.elementtree.rst]: I wouldn't specify the exact exception here, although suggesting it as a common one seems ok.

**Review by scoder — CHANGES_REQUESTED** [review]
Thanks! I left some comments.

**Comment by scoder** [comment]
Please also fix the travis errors. There seems to be trailing whitespace in the file.

File: Doc/library/xml.etree.elementtree.rst (hunk 1 above)
Comment (bansalanjali2512) [reply Doc/library/xml.etree.elementtree.rst]: Thanks @scoder for the valuable feedback. I'll update all of them.

**Comment by scoder** [comment]
Any update on this? It would be nice to get this ready to be merged.

File: Doc/library/xml.etree.elementtree.rst (hunk 5 above)
Comment (bansalanjali2512) [reply Doc/library/xml.etree.elementtree.rst]: If not given, encoding is 'utf-8'

**Comment by tirkarthi** [comment]
There are some trailing spaces in the PR. Please try running `make patchcheck` locally that will fix the whitespace related errors.

**Review by scoder — APPROVED** [review]

**Pull request merged**